src/main/java/org/apache/maven/shared/archiver/PomPropertiesUtil.java

```diff
@@ -71,25 +64,8 @@ private void createPropertiesFile(Properties properties, Path outputFile, boolea
             return;
         }
 
-        try (PrintWriter pw = new PrintWriter(outputFile.toFile(), StandardCharsets.ISO_8859_1.name());
-                StringWriter sw = new StringWriter()) {
-
-            properties.store(sw, null);
-
-            List<String> lines = new ArrayList<>();
-            try (BufferedReader r = new BufferedReader(new StringReader(sw.toString()))) {
-                String line;
-                while ((line = r.readLine()) != null) {
-                    if (!line.startsWith("#")) {
-                        lines.add(line);
-                    }
-                }
-            }
-
-            Collections.sort(lines);
-            for (String l : lines) {
-                pw.println(l);
-            }
+        try (OutputStream out = Files.newOutputStream(outputFile)) {
+            properties.store(out, null);
```

Comment:
I would not touch this code.
The removal of comments and the ordering is definitely important for reproducible builds.
We could refactor the code to use `BufferedReader.lines()`, filter out comments, sort, and print using the streams api.